backend/src/modules/tournament-tables/infrastructure/repository/inmemory/InMemoryTournamentTablesRepository.ts

```diff
@@ -1,16 +1,23 @@
 import { TournamentTablesRepository } from '../../../core/application/TournamentTablesRepository';
 import { TournamentTable } from '../../../core/domain/TournamentTable';
+import { OptimisticLockingException } from '../../../../../shared/core/application/OptimisticLockingException';
 
 export class InMemoryTournamentTablesRepository implements TournamentTablesRepository {
   private readonly entities: { [tableId: string]: TournamentTable } = {};
 
   async save(tournamentTable: TournamentTable): Promise<void> {
     const tableId = `${tournamentTable.tournamentId}_${tournamentTable.tableNumber.raw}`;
-    this.entities[tableId] = tournamentTable;
+    if ((this.entities[tableId]?.version ?? 0) !== tournamentTable.version) {
+      throw new OptimisticLockingException(tournamentTable.version);
+    }
+    this.entities[tableId] = new TournamentTable({
+      ...tournamentTable,
+      version: tournamentTable.version + 1,
+    });
   }
 
   async saveAll(tournamentTables: TournamentTable[]): Promise<void> {
-    Promise.all(tournamentTables.map((table) => this.save(table)));
+    await Promise.all(tournamentTables.map(async (table) => await this.save(table)));
```

Comment:
Ooo, widac progres teamu :) Mówiłem, że z tym będzie problem :) ? 
Ale raczej
`return Promise.all(tournamentTables.map((table) => this.save(table)))`

to by wystarczylo.

Reply:
Zmieniałam to już chyba z 4 razy, więc wolałam teraz dać więcej dla bezpieczeństwa :D

Reply:
Aaa, tak chyba zrobiłam wcześniej i randomowo padał...

Reply:
Było bez returna: 
`    Promise.all(tournamentTables.map((table) => this.save(table)));`
To mogła być przyczyna?

Reply:
Tak, dokładnie :) Pamiętam, że pisałem o tym w PRze wtedy, ale nie wiedzieliście co jest nie tak - to warto było poczekać, żeby sobie zapamiętać :) Widać progress zespołu :D